frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -482,12 +587,40 @@ function ReaderPaging:onPanRelease(_, ges)
             self.view:PanningStop()
         end
     else
-        self.last_pan_relative_y = 0
-        -- trigger full refresh to clear ghosting generated by previous fast refresh
-        UIManager:setDirty(nil, "full")
+        if self._pan_has_scrolled and self._pan_to_scroll_later ~= 0 then
+            self:onPanningRel(self._pan_to_scroll_later)
+        end
+        self._pan_started = false
+        self._pan_page_states_to_restore = nil
+        UIManager.currently_scrolling = false
+        if self._pan_has_scrolled then
+            self._pan_has_scrolled = false
+            -- Don't do any inertial scrolling if pan events come from
+            -- a mousewheel (which may have itself some inertia)
+            if (ges and ges.from_mousewheel) or not self.ui.scrolling:startInertialScroll() then
+                UIManager:setDirty(self.view.dialog, "partial")
+            end
+        end
     end
 end
 
+function ReaderPaging:onHandledAsSwipe()
+    if self._pan_started then
+        -- Restore original position as this pan we've started handling
+        -- has ended up being a multiswipe or handled as a swipe to open
+        -- top or bottom menus
+        if self._pan_has_scrolled then
+            self.view.page_states = self._pan_page_states_to_restore
+            self:_gotoPage(self.view.page_states[#self.view.page_states].page, "scrolling")
+            UIManager:setDirty(self.view.dialog, "fast")
```

Comment:
I needed a refresh here to force the page to be repainted - after cancelling the scroll when it happens to be a multiswipe - which will probably show something (which might be a full screen History, or a little Skim widget).
Made it use the cheapest "fast", that will probably get promoted to something else depending on somewidgets else.
Should it be something else like "ui" ?

Reply:
Assuming it only trips at the end of a gesture, `ui` makes sense, yes. (And it also appears to be the lowest priority after fast).